Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -189,31 +182,29 @@ private func createResolvedPackages(
     allManifests: [GraphLoadingNode],
     mirrors: DependencyMirrors,
     manifestToPackage: [Manifest: Package],
-    // FIXME: This shouldn't be needed once <rdar://problem/33693433> is fixed.
-    rootManifestSet: Set<Manifest>,
     unsafeAllowedPackages: Set<PackageReference>,
     diagnostics: DiagnosticsEngine
 ) -> [ResolvedPackage] {
-
     // Create package builder objects from the input manifests.
-    let packageBuilders: [ResolvedPackageBuilder] = allManifests.compactMap({ node in
+    let packageBuilders: [ResolvedPackageBuilder] = allManifests.compactMap { node in
         guard let package = manifestToPackage[node.manifest] else {
             return nil
         }
-        let isAllowedToVendUnsafeProducts = unsafeAllowedPackages.contains{ $0.path == package.manifest.url }
+        let isAllowedToVendUnsafeProducts = unsafeAllowedPackages.contains { $0.path == package.manifest.url }
         return ResolvedPackageBuilder(
             package,
             productFilter: node.productFilter,
             isAllowedToVendUnsafeProducts: isAllowedToVendUnsafeProducts
         )
-    })
+    }
 
     // Create a map of package builders keyed by the package identity.
-    let packageMapByIdentity: [String: ResolvedPackageBuilder] = packageBuilders.spm_createDictionary{
-        let identity = PackageReference.computeIdentity(packageURL: $0.package.manifest.url)
-        return (identity, $0)
-    }
-    let packageMapByName: [String: ResolvedPackageBuilder] = packageBuilders.spm_createDictionary{ ($0.package.name, $0) }
+    let packageMapByIdentity: [String: ResolvedPackageBuilder] = Dictionary(uniqueKeysWithValues:
```

Comment:
Could this be a mapping with `PackageIdentity` as key?

Reply:
Eventually, yes. But right now, we use the old / current identity semantics, which come from the `computedName`.

Reply:
Thanks!  I am 110% for keeping the old semantics for now and making incremental changes in future PRs.  We will almost certainly want to make the actual semantic changes be behind a feature flag initially.  But my understanding was that `PackageIdentity` would still keep the same package identity as right now, even if it were spelled as `PackageIdentity` rather than `String`, and that the semantic changes would come later?

Reply:
I think the changes in c3fbf4e08ddd54a24fd24b0900702b3e747ad65a should help clarify the direction of this PR. `PackageIdentity` will be the new semantics for package identity, but it's also responsible for / capable of providing the old / current semantics (`PackageReference.Identity`, which is a typealias of `String`).